src/Services/API/Project/ProjectService.API/Controllers/ProjectController.cs

```diff
@@ -0,0 +1,121 @@
+    {
+        _context = context;
+    }
+
+    /*
+    [HttpPost]
+    [SwaggerOperation(Summary ="Create project", Description = "Create a project")]
+
+    public async Task<ActionResult<ProjectCreateDto>> CreateProject([FromBody] ProjectCreateDto project)
+    {
+        var result = await _mediator.Send(new CreateProject(project));
+        return Ok(result);
+    }
+
+
+    [HttpGet("{projectId:long}")]
+    [SwaggerOperation(Summary = "Get prject by Id", Description = "Get project")]
+    public async Task<ActionResult<IEnumerable<ProjectDto>>> GetProjectById(
+        [Range(1, long.MaxValue), FromRoute] long projectId)
+    {
+        var result = await _mediator.Send(new GetProjectById(projectId));
+
+        return result.Match<ActionResult>(Ok, x => NotFound());
+    }
+
+    [HttpDelete("{projectId:long}")]
+     [SwaggerOperation(Summary = "Delete project by id", Description = "delete project")]
+      public async Task<IActionResult> ProjectDelete(long id)
+     {
+
+     }
+    */
+
+    [HttpPost]
+    [SwaggerOperation(Summary = "Add project", Description = "add project")]
+    public async Task<ActionResult<Project>> PostProject(ProjectCreateDto request)
+    {
+
+            var project = request.Adapt<Project>();
+            _context.Projects.Add(project);
+            await _context.SaveChangesAsync();
+            return CreatedAtAction(nameof(PostProject), new { id = project.Id }, project);
+    }
+
+    [HttpDelete("{id}")]
+    [SwaggerOperation(Summary = "Delete project by id", Description = "delete project")]
+    public async Task<IActionResult> DeleteProject(long id)
+    {
+        var project = await _context.Projects.FindAsync(id);
+        if (project == null)
+        {
+            return NotFound();
+        }
+
+        project.IsSoftDeleted = true;
+        await _context.SaveChangesAsync();
+
+        return NoContent();
+    }
+
+    [HttpGet] 
+    [SwaggerOperation(Summary = "get all projects", Description = "get all projects")] 
+    public async Task<ActionResult<IEnumerable<Project>>> GetProjects()
+    {
+        if (_context == null)
+        {
+            return NotFound();
+        }
+        return await _context.Projects.ToListAsync();
+    }
+
+    [HttpGet("{id}")]
+    [SwaggerOperation(Summary = "get project by id", Description = "get project by id")]
+    public async Task<ActionResult<Project>> GetProjectDetail(long id)
+    {
+        var project = await _context.Projects.FindAsync(id);
+
+        if (project == null)
+        {
+            return NotFound();
+        }
+
+        return project;
+    }
+
+    [HttpPut("{id}")]
+    [SwaggerOperation(Summary = "update project by id", Description = "update project by id")]
+    public async Task<ActionResult<Project>> UpdateProject(long id, ProjectCreateDto project)
+    {
+        var _update =  _context.Projects.FirstOrDefault(p => p.Id == id);
+        if (_update != null)
```

Comment:
try to return from method as soon as something got wrong, it's easier to return NotFound if project is null then do all logic if it is not null